python_transport/wirepas_gateway/protocol/mqtt_wrapper.py

```diff
@@ -0,0 +1,217 @@
+# Wirepas Oy licensed under Apache License, Version 2.0
+#
+# See file LICENSE for full license details.
+#
+import queue
+import socket
+import ssl
+from select import select
+from threading import Thread, current_thread
+from time import sleep
+
+from paho.mqtt import client as mqtt
+
+
+class MQTTWrapper(Thread):
+    """
+    Class to manage the MQTT main thread and be able to share it with other services
+    In this case, it allows to have all the related mqtt activity happening on same thread
+    to avoid any dead lock from mqtt client.
+    """
+
+    def __init__(self, logger, username, password, host, port, secure_auth=True, tlsfile=None,
```

Comment:
The tlsfile variable should be renamed to ca_certs, since it is what it is. 

It will also avoid confusion when passing in the client certificates, which should be exposed with this change. 

Here's what I think should be added to the constructor:

-   certfile=None
-   keyfile=None
-   cert_reqs=ssl.CERT_REQUIRED
-   tls_version=ssl.PROTOCOL_TLS
-   ciphers=None

We also don't seem to have a way of propagating a signal to exit from the main loop.

Reply:
Yes agree.
In this change I'll just update the tlsfile->ca_certs name.
Adding new parameter will be addressed in a different cr